bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/RuleStatusInfo.java

```diff
@@ -12,18 +12,22 @@
  */
 package org.openhab.core.automation;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * This class is used to present status of a rule. The status consists of three parts:
  * The main status, a status detail and a string description.
  *
  * @author Yordan Mihaylov - Initial contribution
  * @author Kai Kreuzer - Refactored to match ThingStatusInfo implementation
  */
+@NonNullByDefault
 public class RuleStatusInfo {
 
-    private RuleStatus status;
-    private RuleStatusDetail statusDetail;
-    private String description;
+    private @NonNullByDefault({}) RuleStatus status;
```

Comment:
Are we sure that `null` is NEVER present (even when deserializing with GSON)?

Reply:
Fine, if it's only used for serialization. The comment in the no-arg constructor is a bit mis-leading: 

```
/**
     * Default constructor for deserialization e.g. by Gson.
     */
    protected RuleStatusInfo() {
    }
```

Reply:
I just learned they are deserialzed using Gson when events are processed in `RuleEventFactory.createRuleStatusInfoEvent`. :slightly_smiling_face: I guess the same class will also be used to serialize the events.

Reply:
Well, then my original comment is correct. We have no control over the payload that reaches that method (it is called by `createEvent` which is a public method of the component). A JSON with a `status` of `null` will not fail during deserialization but potentially lead to an NPE later.

Reply:
I've added some validation just in case.